addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -71,17 +72,105 @@ def test_delivery_cost_gift_card(self):
                 'price_unit': 120.00,
             })]
         })
-        self._apply_promo_code(sale_normal_delivery_charges, gift_card.code)
-        sale_normal_delivery_charges.action_confirm()
+        self._apply_promo_code(order, gift_card.code)
+        order.action_confirm()
+
+        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
+            'default_order_id': order.id,
+            'default_carrier_id': self.free_delivery.id,
+        }))
+        delivery_wizard.save().button_confirm()
+
+        self.assertEqual(order.order_line.filtered('is_delivery').price_total, 0)
+
+
+    def test_free_delivery_cost_with_ewallet(self):
```

Comment:
Here are the values of the different lines:

- A product to deliver: price total of 230
- eWallet: price total of -230
- Free Shipping: price total of 0

Could you make it so the subtotal is still under 100 but above 0 to be sure to not test the case of a sale order with an total amount of 0 as this might trigger some special cases?